src/products/components/ProductListDatagrid/ProductListDatagrid.tsx

```diff
@@ -78,19 +82,35 @@ export const ProductListDatagrid: React.FC<ProductListDatagridProps> = ({
   hasRowHover,
   rowAnchor,
 }) => {
+  const isChannelSelected = !!selectedChannelId;
+  const history = useHistory();
   const intl = useIntl();
   const { theme } = useTheme();
   const datagrid = useDatagridChangeState();
   const { locale } = useLocale();
   const productsLength = getProductRowsLength(disabled, products, disabled);
+  const { filterWindow } = useConditionalFilterContext();
   const handleColumnChange = useCallback(
     (picked: ProductListColumns[]) => {
       onUpdateListSettings("columns", picked.filter(Boolean));
     },
     [onUpdateListSettings],
   );
+
+  const handlePriceClick = (productId: string) => {
+    if (!productId) return;
+
+    history.replace({ search: getPriceClickSearchParams(history.location.search) });
+
+    filterWindow.setOpen(true);
+  };
```

Comment:
issue: you just smuggled here the price-related logic, to the component that is already large and mixing a variety of contexts. You can create hook for it instead and use it here to encapsulate that logic

Reply:
the hook could return for you an adapter as well so the preparation of it also could be moved